ibis/expr/operations.py

```diff
@@ -696,6 +708,14 @@ class Mean(Reduction):
     output_type = rules.scalar_output(_mean_output_type)
 
 
+class Quantile(Reduction):
+
+    input_type = [value,
+                  number(name='quantile', allow_boolean=False),
```

Comment:
This is actually pretty tricky because of the distinction between a generic list of values (whose elements could each be of a different type) and an array which is a list of values all with the same type.

I'll make an issue about this to refactor this part of the code.

In the meantime, you can work around this by creating an additional expression type, maybe `MultipleQuantile` and then in the `def quantile` function try to construct one and if that fails validation construct the other.